### ISSUE TITLE
mingw-w64-x86_64-clang: missing required signature 

### ISSUE THREAD

**Issue by Adrian-Devries** [msys2/msys2#26]
CLang and LLVm can't be updated.

Shell script:
=======

```
#!/bin/bash
export LC_ALL=C
which rm
rm -f -r -d /etc/pacman.d/gnupg
which pacman-key
pacman-key --init
pacman-key --populate msys2
pacman-key --refresh-keys
pacman-key --list-keys
which gpg
gpg --list-keys
which pacman
pacman -Scc
pacman -Syy
pacman -Suu
pacman -Syu
which pacdiff
pacdiff 
```

Log:
====

```
/usr/bin/rm
/usr/bin/pacman-key
gpg: /etc/pacman.d/gnupg/trustdb.gpg: trustdb created
gpg: no ultimately trusted keys found
gpg: Generating pacman keyring master key...
gpg: skipping control `%no-protection' ()
.................+++++
+++++
gpg: key 4DA4454D marked as ultimately trusted
gpg: Done
==> Updating trust database...
gpg: 3 marginal(s) needed, 1 complete(s) needed, PGP trust model
gpg: depth: 0  valid:   1  signed:   0  trust: 0-, 0q, 0n, 0m, 0f, 1u
==> Appending keys from msys2.gpg...
gpg: 3 marginal(s) needed, 1 complete(s) needed, PGP trust model
gpg: depth: 0  valid:   1  signed:   0  trust: 0-, 0q, 0n, 0m, 0f, 1u
==> Locally signing trusted keys in keyring...
  -> Locally signing key D55E7A6D7CE9BA1587C0ACACF40D263ECA25678A...
  -> Locally signing key 123D4D51A1793859C2BE916BBBE514E53E0D0813...
  -> Locally signing key B91BCF3303284BF90CC043CA9F418C233E652008...
  -> Locally signing key 9DD0D4217D75A33B896159E6DA7EF2ABAEEA755C...
==> Importing owner trust values...
gpg: setting ownertrust to 4
gpg: setting ownertrust to 4
gpg: setting ownertrust to 4
gpg: inserting ownertrust of 4
==> Updating trust database...
gpg: 3 marginal(s) needed, 1 complete(s) needed, PGP trust model
gpg: depth: 0  valid:   1  signed:   4  trust: 0-, 0q, 0n, 0m, 0f, 1u
gpg: depth: 1  valid:   4  signed:   3  trust: 0-, 0q, 0n, 4m, 0f, 0u
gpg: depth: 2  valid:   3  signed:   0  trust: 3-, 0q, 0n, 0m, 0f, 0u
gpg: refreshing 8 keys from hkp://pool.sks-keyservers.net
gpg: requesting key 4DA4454D from hkp server pool.sks-keyservers.net
gpg: requesting key CA25678A from hkp server pool.sks-keyservers.net
gpg: requesting key AEEA755C from hkp server pool.sks-keyservers.net
gpg: requesting key 3E0D0813 from hkp server pool.sks-keyservers.net
gpg: requesting key 3E652008 from hkp server pool.sks-keyservers.net
gpg: requesting key A47D45A1 from hkp server pool.sks-keyservers.net
gpg: requesting key 2C51581E from hkp server pool.sks-keyservers.net
gpg: requesting key 4CA56930 from hkp server pool.sks-keyservers.net
gpgkeys: key 98AFAAD92279E526A4B8C3C4E5D9E9B14DA4454D not found on keyserver
gpg: key CA25678A: "Alexey Pavlov (Alexpux) <alexey.pawlow@gmail.com>" 4 new signatures
gpg: key AEEA755C: "Martell Malone (martell) <martellmalone@gmail.com>" 3 new signatures
gpg: key 3E0D0813: "Ray Donnelly (MSYS2 Developer - master key) <mingw.android@gmail.com>" 2 new signatures
gpg: key 3E652008: "Ignacio Casal Quinteiro <icquinteiro@gmail.com>" 2 new signatures
gpg: key A47D45A1: "Alexey Pavlov (Alexpux) <alexpux@gmail.com>" 1 new signature
gpg: key 2C51581E: "Martell Malone (MSYS2 Developer) <martellmalone@gmail.com>" not changed
gpg: key 4CA56930: "Ray Donnelly (MSYS2 Developer) <mingw.android@gmail.com>" 2 new signatures
gpg: Total number processed: 7
gpg:              unchanged: 1
gpg:         new signatures: 14
gpg: keyserver communications error: key not found
gpg: keyserver communications error: bad public key
gpg: 3 marginal(s) needed, 1 complete(s) needed, PGP trust model
gpg: depth: 0  valid:   1  signed:   4  trust: 0-, 0q, 0n, 0m, 0f, 1u
gpg: depth: 1  valid:   4  signed:   3  trust: 0-, 0q, 0n, 4m, 0f, 0u
gpg: depth: 2  valid:   3  signed:   0  trust: 3-, 0q, 0n, 0m, 0f, 0u
gpg: keyserver refresh failed: bad public key
==> ERROR: A specified local key could not be updated from a keyserver.
/etc/pacman.d/gnupg/pubring.gpg
-------------------------------
pub   2048R/4DA4454D 2017-05-21
uid                  Pacman Keyring Master Key <pacman@localhost>

pub   2048R/CA25678A 2014-09-28
uid                  Alexey Pavlov (Alexpux) <alexey.pawlow@gmail.com>
sub   2048R/2BBF340E 2014-09-28

pub   4096R/AEEA755C 2014-10-04
uid                  Martell Malone (martell) <martellmalone@gmail.com>
sub   4096R/2A292C03 2014-10-04

pub   4096R/3E0D0813 2014-09-28
uid                  Ray Donnelly (MSYS2 Developer - master key) <mingw.android@gmail.com>
sub   4096R/8603AA9D 2014-09-28

pub   2048R/3E652008 2014-09-29
uid                  Ignacio Casal Quinteiro <icquinteiro@gmail.com>
sub   2048R/64D62A76 2014-09-29

pub   2048D/A47D45A1 2013-11-11
uid                  Alexey Pavlov (Alexpux) <alexpux@gmail.com>
sub   2048g/31CF7700 2013-11-11

pub   4096R/2C51581E 2015-07-22
uid                  Martell Malone (MSYS2 Developer) <martellmalone@gmail.com>
sub   4096R/282D6707 2015-07-22

pub   4096R/4CA56930 2014-09-28
uid                  Ray Donnelly (MSYS2 Developer) <mingw.android@gmail.com>
sub   4096R/576CF231 2014-09-28

/usr/bin/gpg
/usr/bin/pacman

Cache directory: /var/cache/pacman/pkg/
:: Do you want to remove ALL files from cache? [y/N] removing all files from cache...

Database directory: /var/lib/pacman/
:: Do you want to remove unused repositories? [Y/n] removing unused sync repositories...
:: Synchronizing package databases...
downloading mingw32.db...
downloading mingw32.db.sig...
downloading mingw64.db...
downloading mingw64.db.sig...
downloading msys.db...
downloading msys.db.sig...
:: Starting core system upgrade...
 there is nothing to do
:: Starting full system upgrade...
resolving dependencies...
looking for conflicting packages...

Packages (8) mingw-w64-i686-clang-4.0.0-1  mingw-w64-i686-dlfcn-1.1.1-1  mingw-w64-i686-glib2-2.52.2-1  mingw-w64-i686-llvm-4.0.0-1  mingw-w64-x86_64-clang-4.0.0-1  mingw-w64-x86_64-dlfcn-1.1.1-1  mingw-w64-x86_64-glib2-2.52.2-1  mingw-w64-x86_64-llvm-4.0.0-1

Total Download Size:    167.79 MiB
Total Installed Size:   804.11 MiB
Net Upgrade Size:      -620.16 MiB

:: Proceed with installation? [Y/n] :: Retrieving packages...
downloading mingw-w64-i686-llvm-4.0.0-1-any.pkg.tar.xz...
downloading mingw-w64-i686-clang-4.0.0-1-any.pkg.tar.xz...
downloading mingw-w64-i686-dlfcn-1.1.1-1-any.pkg.tar.xz...
downloading mingw-w64-i686-glib2-2.52.2-1-any.pkg.tar.xz...
downloading mingw-w64-x86_64-llvm-4.0.0-1-any.pkg.tar.xz...
downloading mingw-w64-x86_64-clang-4.0.0-1-any.pkg.tar.xz...
downloading mingw-w64-x86_64-dlfcn-1.1.1-1-any.pkg.tar.xz...
downloading mingw-w64-x86_64-glib2-2.52.2-1-any.pkg.tar.xz...
checking keyring...
checking package integrity...
error: mingw-w64-x86_64-clang: missing required signature
error: failed to commit transaction (invalid or corrupted package)
Errors occurred, no packages were upgraded.
:: Synchronizing package databases...
 mingw32 is up to date
 mingw64 is up to date
 msys is up to date
:: Starting core system upgrade...
 there is nothing to do
:: Starting full system upgrade...
resolving dependencies...
looking for conflicting packages...

Packages (8) mingw-w64-i686-clang-4.0.0-1  mingw-w64-i686-dlfcn-1.1.1-1  mingw-w64-i686-glib2-2.52.2-1  mingw-w64-i686-llvm-4.0.0-1  mingw-w64-x86_64-clang-4.0.0-1  mingw-w64-x86_64-dlfcn-1.1.1-1  mingw-w64-x86_64-glib2-2.52.2-1  mingw-w64-x86_64-llvm-4.0.0-1

Total Installed Size:   804.11 MiB
Net Upgrade Size:      -620.16 MiB

:: Proceed with installation? [Y/n] checking keyring...
checking package integrity...
error: mingw-w64-x86_64-clang: missing required signature
error: failed to commit transaction (invalid or corrupted package)
Errors occurred, no packages were upgraded.
/usr/bin/pacdiff 
```

**Comment by martell** [comment]
I can confirm this is happening to me also.
Looks like the package is just not signed.
Alexey might update this is a few days when he gets a chance.

Be warned llvm 4.0 comes with regressions more so then fixes especially around `float128` for COFF targets.

**Comment by mati865** [comment]
I've enabled float128 for Clang so it works fine with libstdc++. However `__FLOAT128__` isn't defined because it produced malformed code.

TL;DR
Trying to use float128 will cause build error but https://github.com/Alexpux/MINGW-packages/issues/1676 is "fixed".

**Comment by martell** [comment]
Thanks for the update mati. :)

**Comment by elieux** [comment]
Seems to be resolved.
